reactive/kotlinx-coroutines-reactor/src/ReactorContextInjector.kt

```diff
@@ -0,0 +1,28 @@
+package kotlinx.coroutines.reactor
+
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.reactive.ContextInjector
+import org.reactivestreams.Publisher
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.util.context.Context
+import kotlin.coroutines.CoroutineContext
+
+/** @suppress */
+@InternalCoroutinesApi
+class ReactorContextInjector : ContextInjector {
+    /**
+     * Injects all values from the [ReactorContext] entry of the given coroutine context
+     * into the downstream [Context] of Reactor's [Publisher] instances of [Mono] or [Flux].
+     */
+    override fun <T> injectCoroutineContext(publisher: Publisher<T>, coroutineContext: CoroutineContext): Publisher<T> {
+        val reactorContext = coroutineContext[ReactorContext]?.context
```

Comment:
Please, use `?: return this` idiom on this line. There is no need to perform `when` below if there is no context and no need to duplicate `?: Context.empty()` code twice.